docs/core-concepts/bot-sequences/end-criteria.mdx

```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
+{
+    "path": ..., // The path of the game object to wait for
+    "count": ..., // The integer considered by the count rule
+    "countRule": ... // The rule to apply to the count, such as "GreaterThanEqual"
+}
+```
+
+The `countRule` can be one of the following:
+
+* `Zero` - Indicates that the object should not be present. Ignored the count.
+* `NonZero` - Indicates that the object should be present. Ignores the count.
+* `GreaterThanEqual` - Indicates that the object should be present at least the number of times specified by the count.
+* `LessThanEqual` - Indicates that the object should be present at most the number of times specified by the count.
+
+
+**Example: Waiting for the start button to appear**
+
+```json
+{
+    "name": "Perform an action once the start button appears",
+    "description": "Performs an action once the start button appears",
+    "endCriteria":[
+        {
+            "type": "NormalizedPath",
+            "transient": false,
+            "data": {
+                "path": "UI Canvas/Start Button",
+                "count": 1,
+                "countRule": "GreaterThanEqual"
+            }
+        }
+    ],
+    "botAction": {...}
+}
+```
+
+### PartialNormalizedPath
+
+The `PartialNormalizedPath` end criteria waits for a specific game object to either exist or not exist in the scene. It behaves similarly to
+`NormalizedPath`, but supports partial matches of the path string by checking if the path string is a substring of the actual path.
+
+**Data Format**
+
+`type` is set to `PartialNormalizedPath`.
+
+```json
+// The `endCriteria[].data` format
+{
+    "path": ..., // The path of the game object to wait for
+    "count": ..., // The integer considered by the count rule
+    "countRule": ... // The rule to apply to the count, such as "GreaterThanEqual"
+}
+```
+
+The `countRule` can be one of the following:
+
+* `Zero` - Indicates that the object should not be present. Ignored the count.
+* `NonZero` - Indicates that the object should be present. Ignores the count.
+* `GreaterThanEqual` - Indicates that the object should be present at least the number of times specified by the count.
+* `LessThanEqual` - Indicates that the object should be present at most the number of times specified by the count.
+
+
+**Example: Waiting for any part of the player model to appear**
+
+```json
+{
+    "name": "Perform an action once the start button appears",
+    "description": "Performs an action once the start button appears",
+    "endCriteria":[
+        {
+            "type": "NormalizedPath",
+            "transient": false,
+            "data": {
+                "path": "Player",
+                "count": 1,
+                "countRule": "GreaterThanEqual"
+            }
+        }
+    ],
+    "botAction": {...}
+}
+```
+
+### UIPixelHash
+
+The `UIPixelHash` end criteria waits and passes if the pixels on the screen change. This automatically passes if this is the first frame of the test.
```

Comment:
Should add a line about how this is only used for 3rd party UI systems like GameFace

Reply:
Added note